### PR TITLE
Fix for sandbox restore()

### PR DESCRIPTION
**Pull request by jniemin**
Implementation for sandbox mock was not proper (#15, #18). It did not properly restore mongoose model to original state. Here is a test and fix for it

### CODE REVIEW

**Comment by gaguirre** [comment]
Great @jniemin! Let me review it
Also, could you rebase with master? (instead of merge)

File: lib/index.js [hunk 1]
```diff
@@ -36,4 +36,12 @@ var newMock = function mock(object) {
 };
 
 sinon.mock = newMock;
-sinon.sandbox.mock = newMock;
+sinon.sandbox.mock = function(object){
```
Comment (gaguirre) [inline lib/index.js]: Maybe we can name this function `sandboxMock` so it would be easier to debug.

**Comment by jniemin** [comment]
I renamed sandbox function and rebased it properly

**Comment by gaguirre** [comment]
Excellent! Thanks @jniemin

**Pull request merged**